src/encoding.rs

```diff
@@ -124,34 +134,33 @@ mod test {
         assert!(res.is_err());
     }
 
-//    #[test]
-//    fn serialization_works() {
-//        let data = vec![0u8, 187, 61, 11, 250, 0];
-//        let encoded = Base64(data);
-//
-//        let serialized = to_vec(&encoded).unwrap();
-//        let deserialized: Base64 = from_slice(&serialized).unwrap();
-//
-//        assert_eq!(encoded, deserialized);
-//        assert_eq!(data, deserialized.decode());
-//    }
-//
-//    #[test]
-//    fn deserialize_from_valid_string() {
-//        let b64_str = "ALs9C/oA";
-//        // this is the binary behind above string
-//        let expected = vec![0u8, 187, 61, 11, 250, 0];
-//
-//        let serialized = to_vec(&b64_str).unwrap();
-//        let deserialized: Base64 = from_slice(&serialized).unwrap();
-//        assert_eq!(expected, deserialized.decode());
-//    }
-//
-//    #[test]
-//    fn deserialize_from_invalid_string() {
-//        let invalid_str = "**BAD!**";
-//        let serialized = to_vec(&invalid_str).unwrap();
-//        let deserialized = from_slice::<Base64>(&serialized);
-//        assert!(deserialized.is_err());
-//    }
+    #[test]
+    fn serialization_works() {
+        let data = vec![0u8, 187, 61, 11, 250, 0];
+        let encoded = Base64(data);
```

Comment:
When `Base64` is a thin container around `Vec<u8>`, then this is not encoded. It just gives you the possibility to encode.

What about calling this `Binary`, `Binary::from_base64`, `Binary().to_base64()`?

Reply:
Very nice suggestion